src/endpoints/Auth.php

```diff
@@ -367,18 +369,22 @@ public function ssoServiceCallback(Request $request, Response $response)
     {
         /** @var AuthService $authService */
         $authService = $this->container->get('services')->get('auth');
+
         $session = $this->container->get('session');
-        // TODO: Implement a pull method
-        $redirectUrl = $session->get('sso_origin_url');
-        $session->remove('sso_origin_url');
         $mode = $session->get('mode');
+
+        $redirectUrl = $mode == DirectusUserSessionsTableGateway::TOKEN_COOKIE
```

Comment:
@bjgajjar I'm thinking we might want to add `source=app` or something flag, so we can redirect to `/admin` only when it makes sense 🤔